Book management

marijana-codeus opened the pull request.

bulatovicb committed:
Database structure for books

https://matijabojanic.atlassian.net/browse/OL3-17

bulatovicb committed:
Change Author model relation, set isbn as unique

https://matijabojanic.atlassian.net/browse/OL3-17

marijana-codeus committed:
Merge pull request #21 from bulatovicb/task/OL3-3/OL3-17/Database-structure-for-books

Database structure for books

marijana-codeus committed:
Ignore storage/app/public

marijana-codeus committed:
Merge remote-tracking branch 'origin/book-management' into book-management

marijana-codeus committed:
Remove storage/app/public from Git tracking

bulatovicb committed:
Add BookController

bulatovicb committed:
Book creation

https://matijabojanic.atlassian.net/browse/OL3-18

bulatovicb committed:
Book view

https://matijabojanic.atlassian.net/browse/OL3-10

bulatovicb committed:
Book editing

https://matijabojanic.atlassian.net/browse/OL3-11

bulatovicb committed:
Delete book

https://matijabojanic.atlassian.net/browse/OL3-9

bulatovicb committed:
Add function description

https://matijabojanic.atlassian.net/browse/OL3-9

bulatovicb committed:
Book-category

bulatovicb committed:
Add postman collection

https://matijabojanic.atlassian.net/browse/OL3-9

bulatovicb committed:
Create category

https://matijabojanic.atlassian.net/browse/OL3-66

bulatovicb committed:
Category view

https://matijabojanic.atlassian.net/browse/OL3-67

bulatovicb committed:
Category Edit

https://matijabojanic.atlassian.net/browse/OL3-69

bulatovicb committed:
Category view all

https://matijabojanic.atlassian.net/browse/OL3-70

bulatovicb committed:
Category delete

https://matijabojanic.atlassian.net/browse/OL3-68

Ana0312 committed:
Merge pull request #25 from bulatovicb/task/OL3-3/OL3-10/Book-view

Book view

bulatovicb committed:
Add event for deleting image from storage

https://matijabojanic.atlassian.net/browse/OL3-9

bulatovicb committed:
Add Const for images

Add new migration to make fields nullable

https://matijabojanic.atlassian.net/browse/OL3-18

bulatovicb committed:
Add observer

https://matijabojanic.atlassian.net/browse/OL3-9

bulatovicb committed:
Merge branch 'book-management' into task/OL3-3/OL3-18/Book-creation

bulatovicb committed:
Delete event and listener

https://matijabojanic.atlassian.net/browse/OL3-9

Ana0312 committed:
Merge pull request #24 from bulatovicb/task/OL3-3/OL3-18/Book-creation

Book creation

bulatovicb committed:
Delete usage from EventServiceProvider

https://matijabojanic.atlassian.net/browse/OL3-9

bulatovicb committed:
Merge branch 'book-management' into task/OL3-3/OL3-9/Book-deleting

bulatovicb committed:
Delete usage

Delete space

https://matijabojanic.atlassian.net/browse/OL3-9

marijana-codeus committed:
Merge pull request #27 from bulatovicb/task/OL3-3/OL3-9/Book-deleting

Book deleting

bulatovicb committed:
Make fields nullable

https://matijabojanic.atlassian.net/browse/OL3-11

bulatovicb committed:
Array merge

https://matijabojanic.atlassian.net/browse/OL3-3

bulatovicb committed:
Remove eager load for category, publisher and genre

https://matijabojanic.atlassian.net/browse/OL3-3

bulatovicb committed:
Change rules in validator

https://matijabojanic.atlassian.net/browse/OL3-11

Ana0312 commented on app/Http/Controllers/BookController.php:
```diff
@@ -0,0 +1,152 @@
+     */
+    public function create(Request $request)
+    {
+        $validator = Validator::make(request()->all(), array_merge([
```
Ajde ipak pogledaj dokumentaciju za form request https://laravel.com/docs/12.x/validation#form-request-validation. Ljepse je kad se izdvoji ova logika za validaciju u zasebnu klasu, pa onda mozes tu kreirati metod za images validationRules. Prije bih u form requets klasi, nego u Image modelu.

bulatovicb committed:
Merge branch 'book-management' into task/OL3-3/OL3-11/Book-editing

bulatovicb committed:
Create CreateBookRequest

https://matijabojanic.atlassian.net/browse/OL3-3

bulatovicb committed:
Create Genre management

https://matijabojanic.atlassian.net/browse/OL3-34

bulatovicb committed:
Add Genre Controller

bulatovicb committed:
Genre creation

https://matijabojanic.atlassian.net/browse/OL3-47

bulatovicb committed:
Genre View

https://matijabojanic.atlassian.net/browse/OL3-48

bulatovicb committed:
Add description on create function

https://matijabojanic.atlassian.net/browse/OL3-47

bulatovicb committed:
Genre - View all paginated

https://matijabojanic.atlassian.net/browse/OL3-51

marijana-codeus committed:
Merge pull request #36 from bulatovicb/OL3-48

Genre View

bulatovicb committed:
Merge branch 'OL3-34' into OL3-51

bulatovicb committed:
Merge branch 'OL3-34' into OL3-47

bulatovicb committed:
Limit the number of characters

https://matijabojanic.atlassian.net/browse/OL3-66

bulatovicb committed:
Merge branch 'book-management' into task/OL3-3/OL3-11/Book-editing

marijana-codeus committed:
Merge pull request #26 from bulatovicb/task/OL3-3/OL3-11/Book-editing

Book editing

marijana-codeus committed:
Merge pull request #35 from bulatovicb/OL3-47

Genre Creation

bulatovicb committed:
Merge branch 'OL3-34' into OL3-51

marijana-codeus committed:
Merge pull request #37 from bulatovicb/OL3-51

Genre - View all paginated

bulatovicb committed:
Genre edit

https://matijabojanic.atlassian.net/browse/OL3-49

bulatovicb committed:
Add closing curly brace

bulatovicb committed:
Merge branch 'book-management' into task/OL3-34/Genres-management

bulatovicb committed:
Genre delete

https://matijabojanic.atlassian.net/browse/OL3-50

marijana-codeus committed:
Merge pull request #45 from bulatovicb/task/OL3-50/Genre-delete

Genre delete

bulatovicb committed:
Merge branch 'OL3-34' into task/OL3-49/Genre-edit

bulatovicb committed:
Import-book-data

https://matijabojanic.atlassian.net/browse/OL3-71

bulatovicb committed:
Merge branch 'task/OL3-34/Genres-management' into task/OL3-49/Genre-edit

marijana-codeus committed:
Merge pull request #48 from bulatovicb/task/OL3-49/Genre-edit

Task/ol3 49/genre edit

marijana-codeus committed:
Merge pull request #44 from bulatovicb/task/OL3-34/Genres-management

genres management

bulatovicb committed:
Add ProcessGoogleBooks job

bulatovicb committed:
Import Book Data

https://matijabojanic.atlassian.net/browse/OL3-71

bulatovicb committed:
Add Limiter

bulatovicb committed:
Add parameters in create function

bulatovicb committed:
Add parameters in create function

bulatovicb committed:
Dispatch-batches

https://matijabojanic.atlassian.net/browse/OL3-73

bulatovicb committed:
Implement rate limiting for jobs

https://matijabojanic.atlassian.net/browse/OL3-72

bulatovicb committed:
Update controller

https://matijabojanic.atlassian.net/browse/OL3-71

bulatovicb committed:
Update Controller

https://matijabojanic.atlassian.net/browse/OL3-71

bulatovicb committed:
Add blade

https://matijabojanic.atlassian.net/browse/OL3-73

bulatovicb committed:
Add blade

https://matijabojanic.atlassian.net/browse/OL3-73

bulatovicb committed:
Merge branch 'task/OL3-71/Import-book-data' into task/OL3-73/Dispatch-batches

bulatovicb committed:
Merge branch 'book-management' into task/OL3-71/Import-book-data

bulatovicb committed:
Merge branch 'task/OL3-71/Import-book-data' into task/OL3-72/Implement-rate-limiting-for-jobs

bulatovicb committed:
Merge branch 'task/OL3-71/Import-book-data' into task/OL3-73/Dispatch-batches

bulatovicb committed:
Book view all

Ana0312 committed:
Merge pull request #41 from bulatovicb/task/OL3-68/Category-delete

Category delete

bulatovicb committed:
Add unique constraint to categories name column

https://matijabojanic.atlassian.net/browse/OL3-22

bulatovicb committed:
Merge branch 'book-management' into task/OL3-3/OL3-22/Book-category

bulatovicb committed:
Add unique validation

https://matijabojanic.atlassian.net/browse/OL3-66

bulatovicb committed:
Merge branch 'task/OL3-3/OL3-22/Book-category' into task/OL3-69/Category-edit

Ana0312 committed:
Merge pull request #28 from bulatovicb/task/OL3-3/OL3-22/Book-category

Book category

bulatovicb committed:
Add when function

https://matijabojanic.atlassian.net/browse/OL3-70

bulatovicb committed:
Merge branch 'task/OL3-3/OL3-22/Book-category' into task/OL3-66/Category-creation

Ana0312 committed:
Merge pull request #38 from bulatovicb/task/OL3-66/Category-creation

Category creation

bulatovicb committed:
Merge branch 'task/OL3-3/OL3-22/Book-category' into task/OL3-67/Category-view

Ana0312 committed:
Merge pull request #39 from bulatovicb/task/OL3-67/Category-view

Category view

bulatovicb committed:
Merge branch 'task/OL3-3/OL3-22/Book-category' into task/OL3-69/Category-edit

Ana0312 committed:
Merge pull request #40 from bulatovicb/task/OL3-69/Category-edit

Category Edit

bulatovicb committed:
Change Index function

bulatovicb committed:
Merge branch 'task/OL3-3/OL3-22/Book-category' into task/OL3-70/Category-view-all

bulatovicb committed:
Update index function

bulatovicb committed:
Merge branch 'task/OL3-70/Category-view-all' of https://github.com/bulatovicb/OnlineLibrary into task/OL3-70/Category-view-all

bulatovicb committed:
Update index function

Ana0312 committed:
Merge pull request #42 from bulatovicb/task/OL3-70/Category-view-all

Category view all

bulatovicb committed:
https://matijabojanic.atlassian.net/browse/OL3-22

bulatovicb committed:
Use when method

bulatovicb committed:
Merge branch 'book-management' into task/View-all

Ana0312 committed:
Merge pull request #55 from bulatovicb/task/OL3-3/OL3-22/Book-category

Book category

bulatovicb committed:
Remove space from functions

Update index function

bulatovicb committed:
Merge branch 'book-management' into task/View-all

bulatovicb committed:
Add unique in CreateBookRequest

bulatovicb committed:
:q
Merge branch 'book-management' of https://github.com/bulatovicb/OnlineLibrary into book-management

bulatovicb committed:
Improve code readability

Ana0312 committed:
Merge pull request #52 from bulatovicb/task/View-all

Book view all

bulatovicb committed:
Change email message

bulatovicb committed:
Change updateOrCreate method

marijana-codeus committed:
Merge pull request #50 from bulatovicb/task/OL3-73/Dispatch-batches

dispatch batches

bulatovicb committed:
Reformat updateOrCreate method

bulatovicb committed:
Merge branch 'task/OL3-71/Import-book-data' into task/OL3-72/Implement-rate-limiting-for-jobs

marijana-codeus committed:
Merge pull request #51 from bulatovicb/task/OL3-72/Implement-rate-limiting-for-jobs

Implement rate limiting for jobs

marijana-codeus committed:
Merge pull request #49 from bulatovicb/task/OL3-71/Import-book-data

Import book data
